### PR TITLE
Add more tests to ASSERT_STATUS_CHECKED

### PR DESCRIPTION
**Pull request by mrambacher**
db_options_test
options_file_test
auto_roll_logger_test
options_util_test
persistent_cache_test

### CODE REVIEW

File: db/compaction/compaction_job.cc [hunk 1]
```diff
@@ -606,6 +606,10 @@ Status CompactionJob::Run() {
       status = state.status;
       io_s = state.io_status;
       break;
+    } else if (!state.io_status.ok()) {
```
Comment (zhichao-cao) [inline db/compaction/compaction_job.cc:609]: If `state.io_status` is not ok, `state.status` must be not ok. So this else branch should never be executed.

File: db/compaction/compaction_job.cc [hunk 2]
```diff
@@ -1079,7 +1080,7 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
     CompactionIterationStats range_del_out_stats;
     Status s = FinishCompactionOutputFile(status, sub_compact, &range_del_agg,
                                           &range_del_out_stats);
-    if (status.ok()) {
+    if (!s.ok() && status.ok()) {
```
Comment (zhichao-cao) [inline db/compaction/compaction_job.cc:1083]: This check `!s.ok()` is unnecessary.

File: logging/auto_roll_logger.cc [hunk 3]
```diff
@@ -45,8 +45,8 @@ AutoRollLogger::AutoRollLogger(Env* env, const std::string& dbname,
     RollLogFile();
   }
   GetExistingFiles();
-  ResetLogger();
-  if (status_.ok()) {
+  s = ResetLogger();
+  if (s.ok()) {
```
Comment (zhichao-cao) [inline logging/auto_roll_logger.cc:49]: Should we also check `status_ `here?

File: db/compaction/compaction_job.cc (hunk 2 above)
Comment (mrambacher) [reply db/compaction/compaction_job.cc:1083]: The check is necessary to "clear" the checked flag in s.  Otherwise, we may not clear it.

**Review by zhichao-cao — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**